hexapod_dart/include/hexapod_dart/descriptors.hpp

```diff
@@ -123,6 +124,41 @@ namespace hexapod_dart {
         protected:
             std::vector<double> _rotation_traj;
         };
+
+        struct BodyOrientation : public DescriptorBase {
+        public:
+            BodyOrientation() {}
+
+            template <typename Simu, typename robot>
+            void operator()(Simu& simu, std::shared_ptr<robot> rob, const Eigen::Vector6d& init_trans)
+            {
+                // roll-pitch-yaw
+                Eigen::Matrix3d rr = dart::math::expMapRot(rob->rot());
+                Eigen::Matrix3d ro = dart::math::expMapRot({init_trans[0], init_trans[1], init_trans[2]});
+                auto rpy = dart::math::matrixToEulerXYZ(ro.inverse() * rr);
+
+                _roll_vec.push_back(rpy(0));
+                _pitch_vec.push_back(rpy(1));
+                _yaw_vec.push_back(rpy(2));
+            }
+
+            void get(std::vector<double>& results)
+            {
+                double threshold = (_perc_threshold / 100.0) * dart::math::constants<double>::pi();
+                results.clear();
+                results.push_back(std::round(std::count_if(_roll_vec.begin(), _roll_vec.end(), [&threshold](double i) {return i>threshold; }) / double(_roll_vec.size()) * 100.0) / 100.0);
```

Comment:
> What is the computation done here ? I can't figure it

```cpp
std::round((std::count_if(vec.begin(), vec.end(), lambda)/vec.size()*100.0)/100.0
```

- `std::round` you know what it does. So, using `std::round(x*100.0)/100.0` we just round x to 2 decimal points. E.g.: 23,4567 becomes 23,46
- `std::count_if(vec.begin(), vec.end(), lambda)`: Iterates over vec and counts how many instances satisfy lambda.

I hope it's more clear now..